library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -320,7 +324,7 @@ private boolean updateLocked() {
         long now = systemFacade.currentTimeMillis();
 
         Collection<FileDownloadInfo> allDownloads = downloadsRepository.getAllDownloads();
-        updateTotalBytesFor(allDownloads);
+        updateTotalBytesIfNecessaryFor(allDownloads);
```

Comment:
True, but I've done it this way in order to avoid the iteration and the check every single time when only has to happen once.

With this implementation, the first check will happen anyway immediately or at maximum after 1s from the previous check, and we save the iteration and checks for every execution of the `updateLocked`